src/app/modules/browse/browse-controller.js

```diff
@@ -124,14 +126,26 @@
             }
             return _localize(message, params);
         }
+        function afterCount(count) {
+            $scope.bookCount = $scope.bookCountObject.bookCount = count;
+            $scope.bookMessage = getBookMessage(count);
+            $scope.setPage = function () {};
+            $scope.initialized = true;
+        }
         $scope.getFilteredBookCount = function() {
-            bookService.getFilteredBooksCount($scope.searchText, $scope.shelf, $scope.lang, $scope.tag).then(function (count) {
-                $scope.bookCount = $scope.bookCountObject.bookCount = count;
-                $scope.bookMessage = getBookMessage(count);
-                $scope.setPage = function () {
-                };
-                $scope.initialized = true;
-            });
+            var promise = bookService.getFilteredBooksCount($scope.searchText, $scope.shelf, $scope.lang, $scope.tag, false, true);
+            if ($scope.allLicenses) {
+                promise.then(function (count) {
+                    afterCount(count);
+                });
+            } else {
+                promise.then(function (fullCount) {
+                    bookService.getFilteredBooksCount($scope.searchText, $scope.shelf, $scope.lang, $scope.tag, false, false).then(function (ccCount) {
+                        $scope.numHiddenBooks = fullCount - ccCount;
```

Comment:
I suppose we don't need to initialize this in the other branch? It will default to zero (even if we're switching back from hiding some to showing all)?